### ISSUE TITLE
Insert fails with SQL Error with mariadb 3.2.3 on Linux. Same code works with 2.5.6 on Linux and works with 3.2.3 on Windows

### ISSUE THREAD

**Issue by markddrake** [mariadb-corporation/mariadb-connector-nodejs#274]
Given the following code

```
root@47a34bcff7c1:/usr/src/YADAMU# cat insertIssue3.js
"use strict"
import mariadb from 'mariadb';
import fs from 'fs';

async function main() {

    const connectionDetails = {
            host      : "MARIA11-01"
           ,user      : "root"
           ,password  : "oracle"
           ,port      : 3306
           ,database  : "mysql"
           ,multipleStatements: true
           ,typeCast:true
                   ,bigNumberStrings:true
                   ,dateStrings:true
                   ,rowsAsArray:true}

  let results;

  try {

    const pool = mariadb.createPool(connectionDetails);
    const conn = await pool.getConnection();
    results = await conn.query(`SET AUTOCOMMIT = 0, TIME_ZONE = '+00:00',SESSION INTERACTIVE_TIMEOUT = 600000, WAIT_TIMEOUT = 600000, SQL_MODE='ANSI_QUOTES,PAD_CHAR_TO_FULL_LENGTH', GROUP_CONCAT_MAX_LEN = 1024000, GLOBAL LOCAL_INFILE = 'ON';`);
    results = await conn.query(`drop schema if exists "YADAMU_TEST";`);
    results = await conn.query(`create schema "YADAMU_TEST";`);


    results = await conn.query(`USE "YADAMU_TEST"`);
    results = await conn.query(`create table  if not exists "YADAMU_TEST"."character_types"(
  "character_col" char(1) ,"character_varying_col" longtext ,"character_max_col" mediumtext ,"character_varying_max_col" mediumtext ,"character_4000_col" varchar(4000) ,"character_varying_4000_col" varchar(4000) ,"character_64k_col" text ,"character_varying_64k_col" text ,"text_col" longtext)`);

    const testData = fs.readFileSync('TEST_DATA.json');
        const data = JSON.parse(testData)
        console.log('ROWS',data.length)

        const sql = `insert into "YADAMU_TEST"."character_types" ("character_col","character_varying_col","character_max_col","character_varying_max_col","character_4000_col","character_varying_4000_col","character_64k_col","character_varying_64k_col","text_col") values  (?,?,?,?,?,?,?,?,?)`;

    try {

          results = await conn.beginTransaction()
          results = await conn.query(`SAVEPOINT YADAMU_INSERT; `)

      console.log(1)
      for (let row =0; row < data.length; row++) {
        const nextRow = data[row]
        const results = await conn.query(sql,nextRow);
            console.log(2)
          }
          console.log(3)
          results = await conn.commit()
    } catch (e) {
          console.log(4)
      console.log(e)
    }

   conn.end()
   pool.end();
 } catch(e) {
   console.log(e);
 }
}

main().then(() => {console.log("Success")}).catch((e) => {console.log(e)})
```

Running with node 3.2.3 on Linux results in

```
root@47a34bcff7c1:/usr/src/YADAMU# node -v
v21.6.1
root@47a34bcff7c1:/usr/src/YADAMU# npm ls
yadamu@1.0.0 /usr/src/YADAMU
+-- @aws-sdk/client-s3@3.507.0
+-- @aws-sdk/lib-storage@3.507.0
+-- @azure/storage-blob@12.17.0
+-- @electron/remote@2.1.2
+-- bootstrap-icons@1.11.3
+-- bootstrap@5.3.2
+-- cookie-parser@1.4.6
+-- csv-parser@3.0.0
+-- electron-packager@17.1.2
+-- electron@27.3.1
+-- express-session@1.18.0
+-- express@4.18.2
+-- font-awesome@4.7.0
+-- ibm_db_electron@npm:ibm_db@3.2.3
+-- ibm_db@3.2.3
+-- install@0.13.0
+-- jquery@3.7.1
+-- mariadb@3.2.3
+-- mime-types@2.1.35
+-- mongodb@6.3.0
+-- mssql@10.0.2
+-- mysql@2.18.1
+-- npm@10.4.0
+-- oracledb@6.3.0
+-- pg-copy-streams@6.0.6
+-- pg-query-stream@4.5.3
+-- pg@8.11.3
+-- readable-stream@4.5.2
+-- snowflake-sdk@1.9.3
+-- uuid@9.0.1
`-- wkx@0.5.0

root@47a34bcff7c1:/usr/src/YADAMU# node insertIssue3.js
ROWS 4
1
4
SqlError: (conn=20787, no: 1064, SQLState: 42000) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'CD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD...' at line 1
sql: insert into "YADAMU_TEST"."character_types" ("character_col","character_varying_col","character_max_col","character_varying_max_col","character_4000_col","character_varying_4000_col","character_64k_col","character_varying_64k_col","text_col") values  (?,?,...
    at module.exports.createError (/usr/src/YADAMU/node_modules/mariadb/lib/misc/errors.js:64:10)
    at PacketNodeEncoded.readError (/usr/src/YADAMU/node_modules/mariadb/lib/io/packet.js:589:19)
    at Query.readResponsePacket (/usr/src/YADAMU/node_modules/mariadb/lib/cmd/parser.js:62:28)
    at PacketInputStream.receivePacketBasic (/usr/src/YADAMU/node_modules/mariadb/lib/io/packet-input-stream.js:85:9)
    at PacketInputStream.onData (/usr/src/YADAMU/node_modules/mariadb/lib/io/packet-input-stream.js:135:20)
    at Socket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
    at Readable.push (node:internal/streams/readable:390:5)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
  sqlMessage: "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'CD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD-ABCD...' at line 1",
  sql: 'insert into "YADAMU_TEST"."character_types" ("character_col","character_varying_col","character_max_col","character_varying_max_col","character_4000_col","character_varying_4000_col","character_64k_col","character_varying_64k_col","text_col") values  (?,?,...',
  fatal: false,
  errno: 1064,
  sqlState: '42000',
  code: 'ER_PARSE_ERROR'
}
Success
root@47a34bcff7c1:/usr/src/YADAMU#
```

Reverting to 2.5.6 

```
root@47a34bcff7c1:/usr/src/YADAMU# npm install mariadb@2.5.6

added 2 packages, changed 1 package, and audited 895 packages in 736ms

112 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
root@47a34bcff7c1:/usr/src/YADAMU# node -v
v21.6.1
root@47a34bcff7c1:/usr/src/YADAMU# npm ls
yadamu@1.0.0 /usr/src/YADAMU
+-- @aws-sdk/client-s3@3.507.0
+-- @aws-sdk/lib-storage@3.507.0
+-- @azure/storage-blob@12.17.0
+-- @electron/remote@2.1.2
+-- bootstrap-icons@1.11.3
+-- bootstrap@5.3.2
+-- cookie-parser@1.4.6
+-- csv-parser@3.0.0
+-- electron-packager@17.1.2
+-- electron@27.3.1
+-- express-session@1.18.0
+-- express@4.18.2
+-- font-awesome@4.7.0
+-- ibm_db_electron@npm:ibm_db@3.2.3
+-- ibm_db@3.2.3
+-- install@0.13.0
+-- jquery@3.7.1
+-- mariadb@2.5.6
+-- mime-types@2.1.35
+-- mongodb@6.3.0
+-- mssql@10.0.2
+-- mysql@2.18.1
+-- npm@10.4.0
+-- oracledb@6.3.0
+-- pg-copy-streams@6.0.6
+-- pg-query-stream@4.5.3
+-- pg@8.11.3
+-- readable-stream@4.5.2
+-- snowflake-sdk@1.9.3
+-- uuid@9.0.1
`-- wkx@0.5.0

root@47a34bcff7c1:/usr/src/YADAMU# node insertIssue3.js
ROWS 4
1
2
2
2
2
3
Success
root@47a34bcff7c1:/usr/src/YADAMU#
```

Running on Windows.

```
C:\Development\YADAMU\src>node -v
v20.7.0

C:\Development\YADAMU\src>npm ls
yadamu@1.0.0 C:\Development\YADAMU\src
+-- @aws-sdk/client-s3@3.507.0
+-- @aws-sdk/lib-storage@3.507.0
+-- @azure/storage-blob@12.17.0
+-- @electron/remote@2.1.2
+-- bootstrap-icons@1.11.3
+-- bootstrap@5.3.2
+-- cookie-parser@1.4.6
+-- csv-parser@3.0.0
+-- electron-packager@17.1.2
+-- electron@27.3.1
+-- express-session@1.18.0
+-- express@4.18.2
+-- font-awesome@4.7.0
+-- ibm_db_electron@npm:ibm_db@3.2.3
+-- ibm_db@3.2.3
+-- install@0.13.0
+-- jquery@3.7.1
+-- mariadb@3.2.3
+-- mime-types@2.1.35
+-- mongodb@6.3.0
+-- mssql@10.0.2
+-- mysql@2.18.1
+-- npm@10.4.0
+-- oracledb@6.3.0
+-- pg-copy-streams@6.0.6
+-- pg-query-stream@4.5.3
+-- pg@8.11.3
+-- readable-stream@4.5.2
+-- snowflake-sdk@1.9.3
+-- uuid@9.0.1
`-- wkx@0.5.0


C:\Development\YADAMU\src>node scratch\mariadb\insertIssue3.js
ROWS 4
1
2
2
2
2
3
Success

C:\Development\YADAMU\src>
```

**Comment by markddrake** [comment]
Test Data
[TEST_DATA.zip](https://github.com/mariadb-corporation/mariadb-connector-nodejs/files/14186661/TEST_DATA.zip)

**Comment by rusher** [comment]
It seems similar to #273, i couldn't reproduced the issue either on windows or unix with the same node.js version. 
could you send aditional results like #273 (max_allowed_packet value and debug information) to permit the issue identification

**Comment by rusher** [comment]
in fact, reproduced. This randomly fails

**Comment by markddrake** [comment]
Does this fix all 3 of the issues I reported. When will this be available. Can I just pick up the changed file and drop it into my 3.2.3 installation ?

**Comment by markddrake** [comment]
I tried replacing packet-output-stream.js in my 3.2.3 library

I got the following crash

```
2024-03-12T17:08:47.730Z [INFO][ColdRoomTemperatures_Archive][Batch]: Rows committed: 1100000.
Stacktrace:
    ptr1=0x1482e3a54081
    ptr2=(nil)
    ptr3=(nil)
    ptr4=(nil)
    ptr5=(nil)
    ptr6=(nil)
    failure_message_object=0x7fff419402a0

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x7257d36653f6]
    1: writeLengthEncodedString(aka writeDefaultLengthEncodedString) [0xd5193101571] [/usr/src/YADAMU/node_modules/mariadb/lib/io/bulk-packet.js:~319] [pc=0x7257b39bbe88](this=0x388ab8e57df9 <BulkPacket map = 0x8fdd74f8aa9>,0x1482e3a54081 <String[12959]: u#...<truncated>>>)
    2: writeParam [0xd51931052d1] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/common-binary-cmd.js:~21] [pc=0x7257b39cde0a](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>,0x388ab8e57df9 <BulkPacket map = 0x8fdd74f8aa9>,0x1482e3a54081 <String[12959]: u#...<truncated>>>,0x2bbeaf1c3b81 <EventEmitter map = 0x2db7db522309>,0x2bbeaf1c94c1 <ConnectionInformation map = 0xc99ffcd0ca9>)
    3: sendQueries [0x5e93be343b9] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/batch-bulk.js:~79] [pc=0x7257b39c9e1a](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>)
    4: start [0x5e93be34381] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/batch-bulk.js:~25] [pc=0x7257b390d9ed](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>,0x2bbeaf1ca279 <PacketOutputStream map = 0x3db705977f9>,0x2bbeaf1c3b81 <EventEmitter map = 0x2db7db522309>,0x2bbeaf1c94c1 <ConnectionInformation map = 0xc99ffcd0ca9>)
    5: _addCommandEnablePipeline(aka _addCommandEnablePipeline) [0x2bbeaf1d6cf9] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~1238] [pc=0x7257b39d7b82](this=0x329f7e1c03e1 <undefined>,0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>)
    6: /* anonymous */(aka /* anonymous */) [0x388ab8e57601] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~216] [pc=0x7257b3a73f81](this=0x329f7e1c03e1 <undefined>,0x388ab8e576a9 <JSFunction (sfi = 0x329f7e1d3a31)>,0x388ab8e576e1 <JSFunction (sfi = 0x329f7e1d3a71)>)
    7: new Promise(aka Promise) [0x3225e0a72ca9](this=0x329f7e1c0411 <the_hole_value>,0x388ab8e57601 <JSFunction (sfi = 0x86874e456d9)>)
    8: ConstructFrame [pc: 0x7257d35c9ecf]
    9: batch [0x3c9d352994f1] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~189] [pc=0x7257b3a7eb2a](this=0x2bbeaf1dcd99 <Connection map = 0x224d4ba0c01>,0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">,0x33678fa6a6f9 <JSArray[10000]>)
   10: batch [0x5e93be38321] [file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbDBI.js:~266] [pc=0x7257b3986901](this=0x0d46a2570b81 <EventEmitter map = 0x247f49275b9>,0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">,0x33678fa6a6f9 <JSArray[10000]>)
   11: _writeBatch [0x5e93be3bbe9] [file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbWriter.js:~73] [pc=0x7257b39767c2](this=0x3c1a0fbe0dd1 <Writable map = 0x29338f14ef61>,0x33678fa6a6f9 <JSArray[10000]>,10000)
   12: /* anonymous */ [0x1a7b9c793fa1](this=0x3d6c1ac81171 <JSGlobalProxy>,0x329f7e1c03e1 <undefined>)
   13: StubFrame [pc: 0x7257d36d5d31]
   14: StubFrame [pc: 0x7257d35f74e8]
   15: EntryFrame [pc: 0x7257d35caa43]
   16: ExitFrame [pc: 0x7257d35cec3e]
   17: processTicksAndRejections [0x3d6c1ac86f91] [node:internal/process/task_queues:~67] [pc=0x7257b39ab365](this=0x3d6c1ac828e1 <process map = 0x60b736fd271>)
   18: InternalFrame [pc: 0x7257d35cab1c]
   19: EntryFrame [pc: 0x7257d35ca843]

==== Details ================================================

[0]: ExitFrame [pc: 0x7257d36653f6]
[1]: writeLengthEncodedString(aka writeDefaultLengthEncodedString) [0xd5193101571] [/usr/src/YADAMU/node_modules/mariadb/lib/io/bulk-packet.js:~319] [pc=0x7257b39bbe88](this=0x388ab8e57df9 <BulkPacket map = 0x8fdd74f8aa9>,0x1482e3a54081 <String[12959]: u#...<truncated>>>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function writeDefaultLengthEncodedString(str) {\x0a    //javascript use UCS-2 or UTF-16 string internal representation\x0a    //that means that string to byte will be a maximum of * 3\x0a    // (4 bytes utf-8 are represented on 2 UTF-16 characters)\x0a    if (str.length * 3 + 10 < this.buf.length - this.pos) {\x0a      //reserve position for length indi...

-----------------------------------------
}
[2]: writeParam [0xd51931052d1] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/common-binary-cmd.js:~21] [pc=0x7257b39cde0a](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>,0x388ab8e57df9 <BulkPacket map = 0x8fdd74f8aa9>,0x1482e3a54081 <String[12959]: u#...<truncated>>>,0x2bbeaf1c3b81 <EventEmitter map = 0x2db7db522309>,0x2bbeaf1c94c1 <ConnectionInformation map = 0xc99ffcd0ca9>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function writeParam(out, value, opts, info) {\x0a    let flushed = false;\x0a    switch (typeof value) {\x0a      case 'boolean':\x0a        flushed = out.writeInt8(0x00);\x0a        flushed = out.writeInt8(value ? 0x01 : 0x00) || flushed;\x0a        break;\x0a      case 'bigint':\x0a      case 'number':\x0a        flushed = out.writeInt8(0x00)...

-----------------------------------------
}
[3]: sendQueries [0x5e93be343b9] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/batch-bulk.js:~79] [pc=0x7257b39c9e1a](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function sendQueries() {\x0a    let flushed = false;\x0a    while (!flushed && this.sending && this.valueIdx < this.values.length) {\x0a      this.valueRow = this.values[this.valueIdx++];\x0a\x0a      //********************************************\x0a      // send params\x0a      //********************************************\x0a      const l...

-----------------------------------------
}
[4]: start [0x5e93be34381] [/usr/src/YADAMU/node_modules/mariadb/lib/cmd/batch-bulk.js:~25] [pc=0x7257b390d9ed](this=0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>,0x2bbeaf1ca279 <PacketOutputStream map = 0x3db705977f9>,0x2bbeaf1c3b81 <EventEmitter map = 0x2db7db522309>,0x2bbeaf1c94c1 <ConnectionInformation map = 0xc99ffcd0ca9>) {
  // expression stack (top to bottom)
  [11] : 0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>
  [10] : 0x329f7e1c03e1 <undefined>
  [09] : 0x329f7e1c03e1 <undefined>
  [08] : 0x329f7e1c03e1 <undefined>
  [07] : 0x017135ff3f89 <JSArray[6]>
  [06] : 0x2bbeaf1ca279 <PacketOutputStream map = 0x3db705977f9>
  [05] : 0x329f7e1c18a1 <true>
  [04] : 0x05e93be340a9 <JSFunction BulkPacket (sfi = 0x23af5900dbe9)>
  [03] : 0x05e93be343b9 <JSFunction sendQueries (sfi = 0x23af5900cc19)>
  [02] : 0x329f7e1c03e1 <undefined>
  [01] : 0x329f7e1c03e1 <undefined>
  [00] : 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
--------- s o u r c e   c o d e ---------
function start(out, opts, info) {\x0a    this.sending = true;\x0a    this.info = info;\x0a    this.values = this.initialValues;\x0a\x0a    if (this.opts.timeout) {\x0a      const err = Errors.createError(\x0a        'Cannot use timeout for Batch statement',\x0a        this.sql,\x0a        false,\x0a        info,\x0a        'HY000',\x0a        Error...

-----------------------------------------
}

[5]: _addCommandEnablePipeline(aka _addCommandEnablePipeline) [0x2bbeaf1d6cf9] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~1238] [pc=0x7257b39d7b82](this=0x329f7e1c03e1 <undefined>,0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function _addCommandEnablePipeline(cmd) => {\x0a    cmd.once('send_end', () => {\x0a      setImmediate(_nextSendCmd);\x0a    });\x0a\x0a    _receiveQueue.push(cmd);\x0a    if (_sendQueue.isEmpty()) {\x0a      cmd.start(_out, opts, info);\x0a      if (cmd.sending) {\x0a        _sendQueue.push(cmd);\x0a        cmd.prependOnceListener('send_end', () => {\x0a          ...

-----------------------------------------
}
[6]: /* anonymous */(aka /* anonymous */) [0x388ab8e57601] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~216] [pc=0x7257b3a73f81](this=0x329f7e1c03e1 <undefined>,0x388ab8e576a9 <JSFunction (sfi = 0x329f7e1d3a31)>,0x388ab8e576e1 <JSFunction (sfi = 0x329f7e1d3a71)>) {
  // expression stack (top to bottom)
  [10] : 0x329f7e1c03e1 <undefined>
  [09] : 0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>
  [08] : 0x33678fa6a6f9 <JSArray[10000]>
  [07] : 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
  [06] : 0x2bbeaf1c3b81 <EventEmitter map = 0x2db7db522309>
  [05] : 0x329f7e1c03e1 <undefined>
  [04] : 0x388ab8e576e1 <JSFunction (sfi = 0x329f7e1d3a71)>
  [03] : 0x3d6c1ac82051 <JSFunction Error (sfi = 0x329f7e1d0b31)>
  [02] : 0x2bbeaf1d6cf9 <JSFunction _addCommandEnablePipeline (sfi = 0x2db7db520af1)>
  [01] : 0x388ab8e57719 <EventEmitter map = 0x8fdd74f8839>
  [00] : 0x329f7e1c18a1 <true>
--------- s o u r c e   c o d e ---------
function (resolve, reject) {\x0a      let useBulk = canUseBulk(vals);\x0a\x0a      const cmd = useBulk\x0a        ? new BatchBulk(resolve, reject, _options, opts, _sql, vals)\x0a        : new BatchRewrite(resolve, reject, _options, opts, _sql, vals);\x0a      if (opts.trace) Error.captureStackTrace(cmd);\x0a      _addCommand(cmd...

-----------------------------------------
}

[7]: new Promise(aka Promise) [0x3225e0a72ca9](this=0x329f7e1c0411 <the_hole_value>,0x388ab8e57601 <JSFunction (sfi = 0x86874e456d9)>) {
// optimized frame
--------- s o u r c e   c o d e ---------
<No Source>
-----------------------------------------
}
[8]: ConstructFrame [pc: 0x7257d35c9ecf]
[9]: batch [0x3c9d352994f1] [/usr/src/YADAMU/node_modules/mariadb/lib/connection.js:~189] [pc=0x7257b3a7eb2a](this=0x2bbeaf1dcd99 <Connection map = 0x224d4ba0c01>,0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">,0x33678fa6a6f9 <JSArray[10000]>) {
  // heap-allocated locals
  var _options = 0x329f7e1c03e1 <undefined>
  var _sql = 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
  var vals = 0x33678fa6a6f9 <JSArray[10000]>
  // expression stack (top to bottom)
  [13] : 0x329f7e1c03e1 <undefined>
  [12] : 0x388ab8e57601 <JSFunction (sfi = 0x86874e456d9)>
  [11] : 0x329f7e1c03e1 <undefined>
  [10] : 0x329f7e1c03e1 <undefined>
  [09] : 0x329f7e1c03e1 <undefined>
  [08] : 0x329f7e1c03e1 <undefined>
  [07] : 0x329f7e1c03e1 <undefined>
  [06] : 0x329f7e1c03e1 <undefined>
  [05] : 0x329f7e1c03e1 <undefined>
  [04] : 0x329f7e1c03e1 <undefined>
  [03] : 0x388ab8e57601 <JSFunction (sfi = 0x86874e456d9)>
  [02] : 0x3225e0a72ca9 <JSFunction Promise (sfi = 0x329f7e1d77e1)>
  [01] : 0x2bbeaf1f9069 <FunctionContext[38]>
  [00] : 0x33678fa6a6f9 <JSArray[10000]>
--------- s o u r c e   c o d e ---------
function (sql, initialValues) => {\x0a    let _options,\x0a      _sql,\x0a      _values = initialValues;\x0a    if (typeof sql === 'object') {\x0a      _options = sql;\x0a      _sql = _options.sql;\x0a      if (_options.values) _values = _options.values;\x0a    } else {\x0a      _sql = sql;\x0a    }\x0a\x0a    if (!_values) {\x0a      return Prom...

-----------------------------------------
}

[10]: batch [0x5e93be38321] [file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbDBI.js:~266] [pc=0x7257b3986901](this=0x0d46a2570b81 <EventEmitter map = 0x247f49275b9>,0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">,0x33678fa6a6f9 <JSArray[10000]>) {
  // expression stack (top to bottom)
  [17] : 0x2bbeaf1dcd99 <Connection map = 0x224d4ba0c01>
  [16] : 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
  [15] : 0x33678fa6a6f9 <JSArray[10000]>
  [14] : 0x329f7e1c03e1 <undefined>
  [13] : 0x329f7e1c03e1 <undefined>
  [12] : 0x329f7e1c03e1 <undefined>
  [11] : 0x329f7e1c03e1 <undefined>
  [10] : 0x329f7e1c03e1 <undefined>
  [09] : 0x2bbeaf1dcd99 <Connection map = 0x224d4ba0c01>
  [08] : 0x3c9d352994f1 <JSFunction Connection.batch (sfi = 0x2db7db51f641)>
  [07] : 0x05e93be372b1 <BlockContext[6]>
  [06] : 0x05e93be372b1 <BlockContext[6]>
  [05] : 0x329f7e1c03e1 <undefined>
  [04] : 0x388ab8e56b79 <HeapNumber 188955>
  [03] : 0x329f7e1c03e1 <undefined>
  [02] : 0x329f7e1c18a1 <true>
  [01] : 0x388ab8e575a9 <String[485]: c"Error\n    at MariadbQA.batch (file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbDBI.js:276:11)\n    at MariadbWriter._writeBatch (file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbWriter.js:83:42)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async MariadbWriter.processBatch (file:///usr/src/YADAMU/src/node/dbi/base/yadamuWriter.js:322:26)\n    at async MariadbWriter.doWrite (file:///usr/src/YADAMU/src/node/dbi/base/yadamuWriter.js:400:9)">
  [00] : 0x388ab8e569b9 <JSAsyncFunctionObject>
--------- s o u r c e   c o d e ---------
function batch(sqlStatement,data) {\x0d\x0a     \x0d\x0a    let stack\x0d\x0a    let attemptReconnect = this.ATTEMPT_RECONNECTION;\x0d\x0a    this.SQL_TRACE.traceSQL(sqlStatement)\x0d\x0a    \x0d\x0a\x09while (true) {\x0d\x0a      // Exit with result or exception.  \x0d\x0a      try {\x0d\x0a        const sqlStartTime = performance.now()\x0d\x0a\x09\x09stack = new Error().stack\x0d\x0a\x09...

-----------------------------------------
}

[11]: _writeBatch [0x5e93be3bbe9] [file:///usr/src/YADAMU/src/node/dbi/mariadb/mariadbWriter.js:~73] [pc=0x7257b39767c2](this=0x3c1a0fbe0dd1 <Writable map = 0x29338f14ef61>,0x33678fa6a6f9 <JSArray[10000]>,10000) {
  // expression stack (top to bottom)
  [16] : 0x0d46a2570b81 <EventEmitter map = 0x247f49275b9>
  [15] : 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
  [14] : 0x33678fa6a6f9 <JSArray[10000]>
  [13] : 0x329f7e1c03e1 <undefined>
  [12] : 0x329f7e1c03e1 <undefined>
  [11] : 0x329f7e1c03e1 <undefined>
  [10] : 0x329f7e1c03e1 <undefined>
  [09] : 0x329f7e1c03e1 <undefined>
  [08] : 0x0d46a2570b81 <EventEmitter map = 0x247f49275b9>
  [07] : 0x05e93be38321 <JSFunction batch (sfi = 0x1c76a5f3e0f9)>
  [06] : 0x03db70591a19 <ModuleContext[3]>
  [05] : 0x03db70591a19 <ModuleContext[3]>
  [04] : 0x329f7e1c03e1 <undefined>
  [03] : 0x329f7e1c03e1 <undefined>
  [02] : 0x329f7e1c03e1 <undefined>
  [01] : 0x1a7b9c794149 <String[181]: c"insert into "WWI_Warehouse1"."ColdRoomTemperatures_Archive" ("ColdRoomTemperatureID","ColdRoomSensorNumber","RecordedWhen","Temperature","ValidFrom","ValidTo") values  (?,?,?,?,?,?)">
  [00] : 0x1a7b9c794001 <JSAsyncFunctionObject>
--------- s o u r c e   c o d e ---------
function _writeBatch(batch,rowCount) {\x0d\x0a\x09  \x0d\x0a    // console.log(batch[0])\x0d\x0a\x0d\x0a    const sqlStatement = `${this.tableInfo.dml} ${this.tableInfo.rowConstructor}`\x0d\x0a        \x0d\x0a    switch (this.tableInfo.insertMode) {\x0d\x0a      case 'Batch':\x0d\x0a        try {    \x0d\x0a          await this.dbi.createSavePoint();\x0d\x0a          const results...

-----------------------------------------
}

[12]: /* anonymous */ [0x1a7b9c793fa1](this=0x3d6c1ac81171 <JSGlobalProxy>,0x329f7e1c03e1 <undefined>) {
// optimized frame
--------- s o u r c e   c o d e ---------
<No Source>
-----------------------------------------
}
[13]: StubFrame [pc: 0x7257d36d5d31]
[14]: StubFrame [pc: 0x7257d35f74e8]
[15]: EntryFrame [pc: 0x7257d35caa43]
[16]: ExitFrame [pc: 0x7257d35cec3e]
[17]: processTicksAndRejections [0x3d6c1ac86f91] [node:internal/process/task_queues:~67] [pc=0x7257b39ab365](this=0x3d6c1ac828e1 <process map = 0x60b736fd271>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function processTicksAndRejections() {\x0a  let tock;\x0a  do {\x0a    while ((tock = queue.shift()) !== null) {\x0a      const asyncId = tock[async_id_symbol];\x0a      emitBefore(asyncId, tock[trigger_async_id_symbol], tock);\x0a\x0a      try {\x0a        const callback = tock.callback;\x0a        if (tock.args === undefined) {\x0a          callback();\x0a       ...

-----------------------------------------
}
[18]: InternalFrame [pc: 0x7257d35cab1c]
[19]: EntryFrame [pc: 0x7257d35ca843]
=====================
```

Can I get an early access of 3.2.4 since it's more than possible I messed something up.

**Comment by markddrake** [comment]
Please ignore my post from yesterday, now deleted. It was, as I suspected, a configuration issue. Am re-testing now.

**Comment by rusher** [comment]
closing since this must have been corrected in 3.3.0. If that's not the case, feel free to create new issue related to this one
